types/webpack/index.d.ts

```diff
@@ -1040,6 +1043,13 @@ declare namespace webpack {
         writeFile(path: string, data: any, callback: (err: Error) => void): void;
     }
 
+    interface SortableSet<T> extends Set<T> {
```

Comment:
I don't think this is totally correct, unless I am missing something.

`SortableSet` is a interface, but the a set is a `Class`, thus if you try to;

```ts
const x = new SortableSet([]);
```

You will end up with an error `SortableSet' only refers to a type, but is being used as a value here.`

Reply:
I agree with the class vs interface. That said, I guess that brings me to a question around how to treat private APIs in type defs. Webpack doesn't export that util at their top-level (see https://github.com/webpack/webpack/blob/HEAD/lib/webpack.js#L165); in other words, I'm not sure if users are intended to do `new SortableSet([])` at all. That's why I made it an interface vs a class. To import `SortableSet` you'd have to do something like `import {} from 'webpack/lib/util/SortableSet`.

Happy to change to a class—it's a quick change. However, I'm unclear on how to handle due to the above. I could even put `SortableSet` in a `util` namespace, but ultimately you wouldn't actually be importing the class in Javascript. Let me know your thoughts.

Reply:
This cannot be a private since it’s being used by an exported member.

If the sortableset is not meant to be used as a class, I don’t have any problem with the current implementation using an interface.

That said, I am gonna loop in @Hotell, for another option.

Reply:
yo folks! sorry to be kinda late to the party. If `SortableSet` is a runtime entity ( which is -> class ) it should be typed as runtime executable , in our case a es2015 class. Also it should be moved under proper path like within implementation, so consumer wont be able to do `const {SortableSet} = require('webpack')`